### ISSUE TITLE
SPP_INITIATOR_DEMO / SPP_ACCEPTOR_DEMO ASSERT_WARN() when devices moved out of range (IDFGH-5554)

### ISSUE THREAD

**Issue by BenjaminWKidd** [espressif/esp-idf#7277]
## Environment

- Development Kit:     [ESP32-DevKitC|
- Kit version for [v4]
- Module or chip used:  [ESP32-WROOM-32D]
- IDF version: [v4.2]
- Build System:         [CMake]
- Operating System:     [Windows]
- Using an IDE?: [Yes (eclipse)]
- Power Supply:         [USB]

## Problem Description

When SPP_ACCEPTOR_DEMO and SPP_INITIATOR_DEMO examples are loaded onto ESP32 DevKitC boards and the boards are moved to apart until the devices are out of range of one another, various ASSERT_WARN() and bta_dm_pm_btm_status hci_status errors appear. Devices appear to cease operation until either both devices are reset, or device running INITIATOR code is reset twice.

### Expected Behavior

I would expect devices to show disconnection warning and (ideally) reconnect and resume normal operation after they are moved to within range of one another (I realize this may require additional code, though I have made an initial attempt at this in my version of the code included below).

### Actual Behavior

Debug output shows "ASSERT_WARN(51 9), in lc_task.c at line 1298E (5587820) BT_APPL: bta_dm_pm_btm_status hci_status=32" and similar other ASSERT_WARN / bta_dm_pm_btm_status hci_status errors. Once these errors appear, only a hard reset (pushbutton or re-powering) allows them to resume normal operation as described above.

### Steps to reproduce

1. Load EXAMPLE_SPP_ACCEPTOR_DEMO bluetooth classic IDF v4.2 example project and (optionally) overwrite with my version of the EXAMPLE_SPP_ACCEPTOR_DEMO.c file, build and load onto an ESP32-WROOM-32D equipped ESP32-DevKitC board.
2. Do the same with the EXAMPLE_SPP_INITIATOR_DEMO example project.
3. Connect both boards to two different computers (and observe the debug output). Move one of the boards away from the other until previously reported output shows up. Note this distance was around 150 ft thru multiple walls in an indoor environment for me.

### Code to reproduce this issue

<script src="https://gist.github.com/BenjaminWKidd/edc9a463ca89194421b2d0d12adb5031.js"></script>

## Debug Logs (SPP_ACCEPTOR_DEMO)

W (5517460) BT_APPL: new conn_srvc id:26, app_id:255
I (5517470) SPP_ACCEPTOR_DEMO: ESP_SPP_SRV_OPEN_EVT
I (5517720) SPP_ACCEPTOR_DEMO: speed(2283.754639s ~ 2284.009766s): 2824.946533 kbit/s
I (5517730) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:0
I (5518710) SPP_ACCEPTOR_DEMO: speed(2284.009766s ~ 2285.003662s): 1091.703247 kbit/s
I (5518720) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-5
I (5519720) SPP_ACCEPTOR_DEMO: speed(2285.003662s ~ 2286.004883s): 1249.834351 kbit/s
I (5519720) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-6
I (5520710) SPP_ACCEPTOR_DEMO: speed(2286.004883s ~ 2287.003662s): 1189.452026 kbit/s
I (5520720) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-6
I (5521720) SPP_ACCEPTOR_DEMO: speed(2287.003662s ~ 2288.006104s): 1090.298096 kbit/s
I (5521720) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-5
I (5522720) SPP_ACCEPTOR_DEMO: speed(2288.006104s ~ 2289.004883s): 1141.873901 kbit/s
I (5522720) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-5
I (5523720) SPP_ACCEPTOR_DEMO: speed(2289.004883s ~ 2290.009766s): 1205.871948 kbit/s
I (5523720) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-5
I (5524710) SPP_ACCEPTOR_DEMO: speed(2290.009766s ~ 2291.000977s): 1070.690430 kbit/s
I (5524720) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-6
I (5525720) SPP_ACCEPTOR_DEMO: speed(2291.000977s ~ 2292.004883s): 1151.820801 kbit/s
I (5525720) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-6
I (5526710) SPP_ACCEPTOR_DEMO: speed(2292.004883s ~ 2293.003662s): 1229.100342 kbit/s
I (5526720) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-5
I (5527730) SPP_ACCEPTOR_DEMO: speed(2293.003662s ~ 2294.017334s): 1140.724121 kbit/s
I (5527730) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-3
I (5528720) SPP_ACCEPTOR_DEMO: speed(2294.017334s ~ 2295.005859s): 1217.813965 kbit/s
I (5528720) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-2
I (5529710) SPP_ACCEPTOR_DEMO: speed(2295.005859s ~ 2296.002197s): 1009.537048 kbit/s
I (5529720) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-2
I (5530720) SPP_ACCEPTOR_DEMO: speed(2296.002197s ~ 2297.003662s): 1170.445557 kbit/s
I (5530720) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-7
I (5531720) SPP_ACCEPTOR_DEMO: speed(2297.003662s ~ 2298.007080s): 1199.739380 kbit/s
I (5531730) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-13
I (5532710) SPP_ACCEPTOR_DEMO: speed(2298.007080s ~ 2299.003418s): 985.689697 kbit/s
I (5532720) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-16
I (5533720) SPP_ACCEPTOR_DEMO: speed(2299.003418s ~ 2300.005859s): 963.886780 kbit/s
I (5533720) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-16
I (5534720) SPP_ACCEPTOR_DEMO: speed(2300.005859s ~ 2301.003906s): 920.517883 kbit/s
I (5534720) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-18
I (5535710) SPP_ACCEPTOR_DEMO: speed(2301.003906s ~ 2302.000977s): 921.419495 kbit/s
I (5535720) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-21
I (5536710) SPP_ACCEPTOR_DEMO: speed(2302.000977s ~ 2303.003174s): 529.476624 kbit/s
I (5536720) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-23
I (5537730) SPP_ACCEPTOR_DEMO: speed(2303.003174s ~ 2304.015869s): 610.015686 kbit/s
I (5537730) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-27
I (5538720) SPP_ACCEPTOR_DEMO: speed(2304.015869s ~ 2305.006836s): 535.477051 kbit/s
I (5538720) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-31
I (5539720) SPP_ACCEPTOR_DEMO: speed(2305.006836s ~ 2306.007324s): 657.039185 kbit/s
I (5539720) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-32
I (5540720) SPP_ACCEPTOR_DEMO: speed(2306.007324s ~ 2307.006836s): 515.051514 kbit/s
I (5540720) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-32
I (5541710) SPP_ACCEPTOR_DEMO: speed(2307.006836s ~ 2308.000732s): 430.306366 kbit/s
I (5541720) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-32
I (5542730) SPP_ACCEPTOR_DEMO: speed(2308.000732s ~ 2309.018555s): 536.911011 kbit/s
I (5542730) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-35
I (5543740) SPP_ACCEPTOR_DEMO: speed(2309.018555s ~ 2310.024414s): 393.693207 kbit/s
I (5543740) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-36
I (5544710) SPP_ACCEPTOR_DEMO: speed(2310.024414s ~ 2311.000732s): 348.820648 kbit/s
I (5544720) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-36
I (5545730) SPP_ACCEPTOR_DEMO: speed(2311.000732s ~ 2312.017090s): 319.493896 kbit/s
I (5545730) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-39
I (5546790) SPP_ACCEPTOR_DEMO: speed(2312.017090s ~ 2313.081787s): 304.988098 kbit/s
I (5546800) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-37
I (5547840) SPP_ACCEPTOR_DEMO: speed(2313.081787s ~ 2314.130615s): 166.128265 kbit/s
I (5547850) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-39
I (5548790) SPP_ACCEPTOR_DEMO: speed(2314.130615s ~ 2315.075684s): 150.846237 kbit/s
I (5548790) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-41
I (5549730) SPP_ACCEPTOR_DEMO: speed(2315.075684s ~ 2316.019531s): 100.694221 kbit/s
I (5549740) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-41
I (5550730) SPP_ACCEPTOR_DEMO: speed(2316.019531s ~ 2317.015625s): 143.119064 kbit/s
I (5550730) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-42
I (5551710) SPP_ACCEPTOR_DEMO: speed(2317.015625s ~ 2318.000488s): 321.669006 kbit/s
I (5551720) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-42
I (5552750) SPP_ACCEPTOR_DEMO: speed(2318.000488s ~ 2319.039551s): 213.423157 kbit/s
I (5552750) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-43
I (5553870) SPP_ACCEPTOR_DEMO: speed(2319.039551s ~ 2320.161865s): 155.250610 kbit/s
I (5553880) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-42
I (5554740) SPP_ACCEPTOR_DEMO: speed(2320.161865s ~ 2321.028076s): 228.581741 kbit/s
I (5554740) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-44
I (5555790) SPP_ACCEPTOR_DEMO: speed(2321.028076s ~ 2322.074463s): 90.826843 kbit/s
I (5555790) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-43
I (5557620) SPP_ACCEPTOR_DEMO: speed(2322.074463s ~ 2323.908203s): 60.466579 kbit/s
I (5557620) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-44
I (5558520) SPP_ACCEPTOR_DEMO: speed(2323.908203s ~ 2324.804688s): 17.669020 kbit/s
I (5558520) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-46
I (5558800) SPP_ACCEPTOR_DEMO: speed(2324.804688s ~ 2325.085938s): 56.320000 kbit/s
I (5558800) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-47
I (5561410) SPP_ACCEPTOR_DEMO: speed(2325.085938s ~ 2327.703369s): 27.232803 kbit/s
I (5561420) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-47
I (5565960) SPP_ACCEPTOR_DEMO: speed(2327.703369s ~ 2332.248535s): 1.742511 kbit/s
I (5565960) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-49
I (5579810) SPP_ACCEPTOR_DEMO: speed(2332.248535s ~ 2346.099609s): 0.571797 kbit/s
I (5579820) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-49
ASSERT_WARN(51 9), in lc_task.c at line 1298E (5587820) BT_APPL: bta_dm_pm_btm_status hci_status=32

## Debug Logs (SPP_INITIATOR_DEMO)
I (76) boot:  1 phy_init         RF data          01 01 0000f000 00001000
I (83) boot:  2 factory          factory app      00 00 00010000 00100000
I (91) boot: End of partition table
I (95) boot_comm: chip revision: 1, min. application chip revision: 0
I (102) esp_image: segment 0: paddr=0x00010020 vaddr=0x3f400020 size=0x17c98 ( 97432) map
I (148) esp_image: segment 1: paddr=0x00027cc0 vaddr=0x3ffbdb60 size=0x03358 ( 13144) load
I (154) esp_image: segment 2: paddr=0x0002b020 vaddr=0x40080000 size=0x00404 (  1028) load
I (156) esp_image: segment 3: paddr=0x0002b42c vaddr=0x40080404 size=0x04bec ( 19436) load
I (173) esp_image: segment 4: paddr=0x00030020 vaddr=0x400d0020 size=0x647c8 (411592) map
I (330) esp_image: segment 5: paddr=0x000947f0 vaddr=0x40084ff0 size=0x10398 ( 66456) load
I (370) boot: Loaded app from partition at offset 0x10000
I (371) boot: Disabling RNG early entropy source...
I (371) cpu_start: Pro cpu up.
I (375) cpu_start: Application information:
I (379) cpu_start: Project name:     bt_spp_initiator_demo
I (386) cpu_start: App version:      1
I (390) cpu_start: Compile time:     Jul  8 2021 08:10:21
I (396) cpu_start: ELF file SHA256:  f2f912f80220f3bd...
I (402) cpu_start: ESP-IDF:          v4.2-dirty
I (407) cpu_start: Starting app cpu, entry point is 0x40081878
I (0) cpu_start: App cpu up.
I (418) heap_init: Initializing. RAM available for dynamic allocation:
I (425) heap_init: At 3FFAFF10 len 000000F0 (0 KiB): DRAM
I (431) heap_init: At 3FFB6388 len 00001C78 (7 KiB): DRAM
I (437) heap_init: At 3FFB9A20 len 00004108 (16 KiB): DRAM
I (443) heap_init: At 3FFCA550 len 00015AB0 (86 KiB): DRAM
I (449) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM
I (455) heap_init: At 3FFE4350 len 0001BCB0 (111 KiB): D/IRAM
I (462) heap_init: At 40095388 len 0000AC78 (43 KiB): IRAM
I (468) cpu_start: Pro cpu start user code
I (486) spi_flash: detected chip: generic
I (487) spi_flash: flash io: dio
W (487) spi_flash: Detected size(16384k) larger than the size in the binary image header(2048k). Using the size in the binary image header.
I (498) cpu_start: Starting scheduler on PRO CPU.
I (0) cpu_start: Starting scheduler on APP CPU.
I (585) BTDM_INIT: BT controller compile version [f07dfd2]
I (585) system_api: Base MAC address is not set
I (585) system_api: read default base MAC address from EFUSE
I (685) phy: phy_version: 4500, 0cd6843, Sep 17 2020, 15:37:07, 0, 0
I (1235) SPP_INITIATOR_DEMO: ESP_SPP_INIT_EVT
I (1255) SPP_INITIATOR_DEMO: ESP_BT_GAP_DISC_STATE_CHANGED_EVT
I (1605) SPP_INITIATOR_DEMO: ESP_BT_GAP_DISC_RES_EVT
I (1605) SPP_INITIATOR_DEMO: f4 cf a2 bc 5c b2 
I (1605) SPP_INITIATOR_DEMO: AWPL146478
I (2065) SPP_INITIATOR_DEMO: ESP_BT_GAP_DISC_RES_EVT
I (2065) SPP_INITIATOR_DEMO: 98 f4 ab 06 3c ba 
I (2065) SPP_INITIATOR_DEMO: ESP_SPP_ACCEPTOR
I (2065) SPP_INITIATOR_DEMO: ESP_BT_GAP_DISC_RES_EVT
I (2075) SPP_INITIATOR_DEMO: f4 cf a2 bc 5c b2 
I (2085) SPP_INITIATOR_DEMO: ESP_BT_GAP_DISC_STATE_CHANGED_EVT
I (3995) SPP_INITIATOR_DEMO: ESP_SPP_DISCOVERY_COMP_EVT status=0 scn_num=1
I (4005) SPP_INITIATOR_DEMO: ESP_SPP_CL_INIT_EVT
W (4575) BT_APPL: new conn_srvc id:26, app_id:1
I (4585) SPP_INITIATOR_DEMO: ESP_SPP_OPEN_EVT
I (5515) SPP_INITIATOR_DEMO: speed(4.064829s ~ 5.001528s): 904.709045 kbit/s
I (5525) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-8
I (6535) SPP_INITIATOR_DEMO: speed(5.001528s ~ 6.018050s): 1285.559448 kbit/s
I (6545) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-9
I (7515) SPP_INITIATOR_DEMO: speed(6.018050s ~ 7.001550s): 1135.454956 kbit/s
I (7525) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-9
I (8515) SPP_INITIATOR_DEMO: speed(7.001550s ~ 8.001143s): 1228.099487 kbit/s
I (8525) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-8
I (9545) SPP_INITIATOR_DEMO: speed(8.001143s ~ 9.028234s): 1094.977051 kbit/s
I (9555) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-8
I (10515) SPP_INITIATOR_DEMO: speed(9.028234s ~ 10.001188s): 1188.462280 kbit/s
I (10525) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-8
I (11515) SPP_INITIATOR_DEMO: speed(10.001188s ~ 11.001247s): 1013.700073 kbit/s
I (11525) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-9
I (12535) SPP_INITIATOR_DEMO: speed(11.001247s ~ 12.016811s): 1185.390625 kbit/s
I (12545) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-9
I (13555) SPP_INITIATOR_DEMO: speed(12.016811s ~ 13.037048s): 1164.435425 kbit/s
I (13565) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-8
I (14515) SPP_INITIATOR_DEMO: speed(13.037048s ~ 14.000269s): 1208.695068 kbit/s
I (14525) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-5
I (15525) SPP_INITIATOR_DEMO: speed(14.000269s ~ 15.008049s): 1155.251953 kbit/s
I (15535) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-6
I (16515) SPP_INITIATOR_DEMO: speed(15.008049s ~ 16.002888s): 1194.163330 kbit/s
I (16525) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-5
I (17515) SPP_INITIATOR_DEMO: speed(16.002888s ~ 17.000805s): 1150.796875 kbit/s
I (17525) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-11
I (18515) SPP_INITIATOR_DEMO: speed(17.000805s ~ 18.000158s): 1046.116455 kbit/s
I (18525) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-12
I (19595) SPP_INITIATOR_DEMO: speed(18.000158s ~ 19.081970s): 1002.984009 kbit/s
I (19605) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-17
I (20525) SPP_INITIATOR_DEMO: speed(19.081970s ~ 20.006609s): 1130.646973 kbit/s
I (20525) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-20
I (21555) SPP_INITIATOR_DEMO: speed(20.006609s ~ 21.036995s): 830.135498 kbit/s
I (21555) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-22
I (22575) SPP_INITIATOR_DEMO: speed(21.036995s ~ 22.057442s): 1024.492432 kbit/s
I (22585) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-25
I (23605) SPP_INITIATOR_DEMO: speed(22.057442s ~ 23.092016s): 505.251190 kbit/s
I (23615) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-27
I (24585) SPP_INITIATOR_DEMO: speed(23.092016s ~ 24.068705s): 681.158936 kbit/s
I (24595) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-32
I (25625) SPP_INITIATOR_DEMO: speed(24.068705s ~ 25.107212s): 503.337738 kbit/s
I (25635) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-32
I (26525) SPP_INITIATOR_DEMO: speed(25.107212s ~ 26.004673s): 582.443176 kbit/s
I (26525) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-35
I (27615) SPP_INITIATOR_DEMO: speed(26.004673s ~ 27.097008s): 522.037781 kbit/s
I (27625) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-35
I (28645) SPP_INITIATOR_DEMO: speed(27.097008s ~ 28.125986s): 507.999023 kbit/s
I (28645) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-37
I (29515) SPP_INITIATOR_DEMO: speed(28.125986s ~ 29.001211s): 561.044250 kbit/s
I (29525) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-37
I (30585) SPP_INITIATOR_DEMO: speed(29.001211s ~ 30.064480s): 387.333893 kbit/s
I (30585) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-39
I (31545) SPP_INITIATOR_DEMO: speed(30.064480s ~ 31.032160s): 392.857147 kbit/s
I (31555) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-41
I (32725) SPP_INITIATOR_DEMO: speed(31.032160s ~ 32.204498s): 283.740601 kbit/s
I (32725) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-42
I (33635) SPP_INITIATOR_DEMO: speed(32.204498s ~ 33.120911s): 414.835083 kbit/s
I (33645) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-42
I (34515) SPP_INITIATOR_DEMO: speed(33.120911s ~ 34.000416s): 171.096222 kbit/s
I (34525) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-41
I (35865) SPP_INITIATOR_DEMO: speed(34.000416s ~ 35.347431s): 135.232315 kbit/s
I (35875) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-44
I (36815) SPP_INITIATOR_DEMO: speed(35.347431s ~ 36.298412s): 99.938889 kbit/s
I (36825) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-44
I (37555) SPP_INITIATOR_DEMO: speed(36.298412s ~ 37.035988s): 128.854599 kbit/s
I (37555) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-45
I (38605) SPP_INITIATOR_DEMO: speed(37.035988s ~ 38.089123s): 270.734528 kbit/s
I (38615) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-45
I (39515) SPP_INITIATOR_DEMO: speed(38.089123s ~ 39.000095s): 278.208160 kbit/s
I (39525) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-46
I (40535) SPP_INITIATOR_DEMO: speed(39.000095s ~ 40.015327s): 171.625778 kbit/s
I (40535) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-46
I (41575) SPP_INITIATOR_DEMO: speed(40.015327s ~ 41.059662s): 182.010666 kbit/s
I (41585) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-47
I (42745) SPP_INITIATOR_DEMO: speed(41.059662s ~ 42.225822s): 122.247314 kbit/s
I (42745) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-47
I (45635) SPP_INITIATOR_DEMO: speed(42.225822s ~ 45.120819s): 49.243584 kbit/s
I (45645) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-50
E (60245) BT_APPL: bta_dm_pm_btm_status hci_status=12
I (66925) SPP_INITIATOR_DEMO: speed(45.120819s ~ 66.405518s): 4.465180 kbit/s
I (66925) esp_bt_gap_cb(): ESP_BT_GAP_READ_RSSI_DELTA_EVT RSSI:-52
ASSERT_WARN(1 6), in lc_task.c at line 4658E (72135) BT_APPL: bta_dm_pm_btm_status hci_status=12

**Comment by BetterJincheng** [comment]
Hi
I tried to follow your steps to reproduce the scenario but falied. Here is my LOGs:
```c
I (8161) SPP_ACCEPTOR_DEMO: ESP_SPP_SRV_OPEN_EVT
I (10511) SPP_ACCEPTOR_DEMO: speed(7.669840s ~ 10.014300s): 695.904236 kbit/s
I (13501) SPP_ACCEPTOR_DEMO: speed(10.014300s ~ 13.006701s): 632.562256 kbit/s
I (16501) SPP_ACCEPTOR_DEMO: speed(13.006701s ~ 16.004412s): 782.036865 kbit/s
I (19511) SPP_ACCEPTOR_DEMO: speed(16.004412s ~ 19.015560s): 991.595093 kbit/s
I (22491) SPP_ACCEPTOR_DEMO: speed(19.015560s ~ 22.000582s): 894.144287 kbit/s
W (28731) BT_RFCOMM: port_rfc_closed RFCOMM connection in state 2 closed: Peer connection failed (res: 16)
I (28731) SPP_ACCEPTOR_DEMO: ESP_SPP_CLOSE_EVT
```

I just booted two ESP32 and put one into the shielding box to act  as 'out of range'. But it did not cause any error logs.  Any suggestion ?
Thanks!

**Comment by BenjaminWKidd** [comment]
Hi @BetterJincheng,

Unfortunately its been a while since I posted this, so I'll have to dig it back up and see if I can make it happen again. I'm pretty sure if you just take the \esp-idf-2\examples\bluetooth\bluedroid\classic_bt\bt_spp_acceptor and bt_spp_initiator and put enough distance between the two so that they go out of range, that you will see the assert(), but I will have to double check that I can reproduce it. I've since moved on to trying out esp-now for this application.
